src/main/java/com/thealgorithms/datastructures/hashmap/hashing/HashMap.java

```diff
@@ -89,7 +89,9 @@ private Node findKey(Node n, int key) {
         public void delete(int key) {
             if (!isEmpty()) {
                 if (first.getKey() == key) {
-                    first = null;
+                    Node head = first;
+                    first = first.next;
+                    head = null;
```

Comment:
If we do first = first.next directly,  The older value is still pointing to the first. next. We must remove that value so it will not pointing to our pointer. It is a more effective and efficient method. This is the best approach to remove head of LinkedList.

For Example.
Our original array is 1->2->3->null
if we are removing 1 from this LinkedList using first= first.next then our result will be like: 2->3->null and still node 1 is pointing to node 2. We haven't made null to next of the node 1. That's why we are using temp variable to make it as a null.

Reply:
But you are assigning `null` to `head`, not `head.next`

Reply:
@siriak When we assign null to any Node, it means we are assigning null to the entire object. Value and next both are getting null. Suppose we are assigning null to head.next it means in memory, we still have a head node that has value along with the next pointer as a null. Assigning null to the entire node is the best approach for a Linked List.

Reply:
@yanglbme what do you think about that?

Reply:
@siriak you can refer built-in implementation of Linked list in Java or Python. They implemented in the same manner that I shown to you.

Reply:
`head` is a local variable here, so assigning `null` to it doesn't really do anything

Reply:
@yanglbme can you verify this solution once?